### PR TITLE
Deprecate List module, introduce LinkedLists

### PR DESCRIPTION
**Pull request by benharsh**
This PR adds a deprecation warning to the "List" module and introduces a new "LinkedLists" module to take its place, and likewise renames the type from "list" to "LinkedList". We expect to eventually use the name "list" (along with "deque") for a type to replace some of the array-as-vector operations.

Modules and tests have been updated to use the new LinkedList type, and a deprecation test has been added.

Testing:
- [x] local
- [x] gasnet

### CODE REVIEW

File: modules/standard/List.chpl [hunk 1]
```diff
@@ -20,12 +20,14 @@
 /*
   This module provides a simple singly linked list.
 
-  .. note::
+  .. warning::
 
-      This module is expected to change in the future.
+      This module has been deprecated.
```
Comment (ben-albrecht) [inline modules/standard/List.chpl]: Can we point to the `LinkedList` module in the docs?

File: modules/standard/List.chpl [hunk 2]
```diff
@@ -20,12 +20,14 @@
...
  */
 module List {
 
+compilerWarning("Module 'List' has been deprecated in favor of 'LinkedList'");
```
Comment (ben-albrecht) [inline modules/standard/List.chpl]: In an effort to standardize deprecation messages (https://github.com/chapel-lang/chapel/issues/12438) - how do you feel about:

```chpl
compilerWarning("List module is deprecated - please use LinkedList instead");
```

**Comment by vasslitvinov** [comment]
My preference is to make the name capitalized: `LinkedList`.

I think this is our convention for library types that are not all-lowercase like `bigint` or `date`.

To me `linkedList` looks more like the name of a variable.

To avoid the potential conflict with the module name, we could follow the lead of the `Barrier` type, whose module is plural `Barriers`. In this case, the module name would be `LinkedLists`. Allowing us to add a doulbly-linked list to it easily later.

**Comment by ben-albrecht** [comment]
@vasslitvinov  makes a good point.

The closest thing to documented guidelines on this is in https://github.com/chapel-lang/chapel/issues/6698 : 

> Types should be CamelCase or lowercase. User-defined types should generally start with an uppercase letter and be CamelCase (e.g. RandomStream,Timer). Internal types can be lowercase (e.g. int, domain,bigint)

I don't think a linked list counts as an internal type, so I agree that `LinkedList` type and `LinkedLists.chpl` is more consistent.

**Comment by benharsh** [comment]
I've pushed some commits that use LinkedList/LinkedLists and will re-test.

@bradcray : do you have a preference?

**Comment by bradcray** [comment]
No particular preference on naming here.

**Pull request merged**